apps/api/pecanapi-spec.yml

```diff
@@ -358,6 +360,94 @@ paths:
           description: Access forbidden
         '404':
           description: Site(s) not found
+  
+  /api/inputs/:
+    get:
+      tags:
+        - inputs
+      summary: Search for the inputs
+      parameters:
+        - in: query
+          name: model_id
+          description: If provided, returns all inputs for the provided model_id
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: site_id
+          description: If provided, returns all inputs for the provided site_id
```

Comment:
I could also see wanting to filter inputs by format_id (i.e. find all the sites that have data of a specific type) and machine_id

Reply:
Ok...I could add those as well.

Reply:
This also means that we should have an API endpoint to get some details about the formats as well, right? Something like `/api/formats/{format_id}`, which will return information about the specified `format_id`? 

And I think we should also allow users to search for formats based on string matching with the `format_name` & `mimetype`. This could be the `/api/formats/?format_name=xyz&mimetype=abc`

Reply:
If it's not too much trouble, yes it would be nice to search formats. If you return info about a format, make sure to grab the formats_variables, and then match variable.id to name and units. Note that the 'units' in the formats_variables references the units in the file itself, while the units in the variables table is the units of the PEcAn standard, and the formats_variables units might be blank if the files is already in PEcAn standard units. Indeed, we might not want to report the PEcAn standard units in cases where the file isn't in PEcAn standard units because that's liable to confuse users, and just use variables.units to gapfill missing units in formats_variables.

Also be aware that mimetypes are pretty standardized, but format names are largely "made up" and descriptive. So searches on the latter are more likely to be partial rather than exact (e.g. find all formats that mention Ameriflux)

Reply:
Yup...I have created a search functionality for the formats...I will return the `formats_variables` as well according to the way you have mentioned.

Yup...I agree that the names are "made up", & hence I have enabled substring matching based filtering, not only here, but in fact for all endpoints that search based on a name (like for models, PFTs, sites, etc.)